test/unit/apps/gro/controllers/confirm.js

```diff
@@ -2,68 +2,216 @@
   hof: {
-    i18n,
     controllers: {
-      confirm: BaseController
+      confirm: sinon.stub()
+    }
+  },
+  '../../common/services/email': {
+    sendEmail() {
+      const cb = arguments[arguments.length - 1];
+      cb();
+    }
+  },
+  '../steps': {
+    '/step-1': {
+      fields: [
+        'field-1',
+        'field-2'
+      ],
+      locals: {
+        section: 'section-1'
+      }
+    },
+    '/step-2': {
+      fields: [
+        'field-3',
+      ],
+      locals: {
+        section: 'section-1'
+      }
+    },
+    '/step-3': {
+      fields: [
+        'field-4',
+        'field-5'
+      ],
+      locals: {
+        section: 'section-2'
+      }
     }
   },
-  '../../common/models/email': Model
+  '../fields': {
+    'field-1': {},
+    'field-2': {},
+    'field-3': {},
+    'field-4': {},
+    'field-5': {
+      includeInEmail: false
+    }
+  }
 });
 
-describe('apps/gro/controllers/confirm', () => {
+describe('Confirm controller', () => {
+  const req = {
+    sessionModel: {
+      toJSON: sinon.stub(),
+      get: sinon.stub()
+    }
+  };
+  const res = {
+    locals: {
+      partials: {}
+    }
+  };
+  let callback;
+  let controller;
+  let promiseStub;
 
-  describe('.saveValues()', () => {
+  beforeEach(() => {
+    promiseStub = data => new Promise((resolve) => resolve(data));
+    controller = new Controller();
+  });
 
-    let controller;
-    const emailText = 'email@email.com';
-    const emailJson = {'email-text': 'email@email.com'};
-    const req = {
-      sessionModel: {
-        toJSON: sinon.stub().returns(emailJson),
-        get: sinon.stub().returns(emailText)
-      },
-    };
-    const res = {};
-    const callback = sinon.stub();
+  describe('formatStepsForEmail', () => {
+    let data;
+    beforeEach(() => {
+      data = controller.formatStepsForEmail({
+        'field-1': 'value-1',
+        'field-3': 'value-3',
+        'field-4': 'value-4',
+        'field-5': 'value-5',
+      });
+    });
+
+    it('returns an array with 2 entries', () => {
+      data.should.be.an('array').and.have.property('length').and.be.equal(2);
+    });
+
+    it('groups by section', () => {
+      data[0].should.have.property('section').and.be.equal('section-1');
+      data[1].should.have.property('section').and.be.equal('section-2');
+    });
+
+    describe('first section fields', () => {
+      it('is an array with 3 entries', () => {
+        data[0].fields.should.be.an('array').and.have.property('length').and.be.equal(3);
+      });
+
+      it('contains fields from section-1', () => {
+        data[0].fields[0].field.should.be.equal('field-1');
+        data[0].fields[1].field.should.be.equal('field-2');
+        data[0].fields[2].field.should.be.equal('field-3');
+      });
+
+      it('contains values passed for the fields', () => {
+        data[0].fields[0].value.should.be.equal('value-1');
+        expect(data[0].fields[1].value).to.be.undefined;
```

Comment:
should.exist(something); // not undefed or null
should.not.exist(somethingElse); // undefined or null
should.equal(somethingElse, undefined);